### PR TITLE
Fix command not found issue (again) 

### PR DESCRIPTION
**Pull request by derrickmar**
DISCLAIMER: I still need to test more, but this is my hunch of what high-level is currently happening.

**What was happening**
Specifically, when I specify `'mountPath': '/home',` for the user container (see jupyterhub_config.py of Kubespawner) and my WORKDIR in my container's Dockerfile specifies any subdirectory of the mountPath (i.e. `WORKDIR /home/jovyan`) , the container fails to startup. It appears to be that it cannot find any executables since I get errors such as `Container command <BLANK>  not found or does not exist`. Additionally, this issue only happens when the persistent volume claim doesn't exist and is created/assigned.

**Why it was happening**
The mounting happens at runtime AFTER all the commands in the Docker container which occur at docker-build time (except for the CMD one). So these subdirectories of '/home' that were created at docker-build time get hidden due to the mounting to the pvc. We were running commmands in a hidden directory which is why I think we couldn't find any executables.

**The temporary fix**
Make sure the WORKDIR command is NOT a subdirectory of the mountPath specified in jupyterhub_config.py. 

**Ideal fix**
Ideally, we want the WORKDIR to /home/[username] because the user should really only be adding files from the notebook server browser in /home/[username]. But since this is a subdirectory of the mount path we'll have to figure out how to do that. The general idea is that we have to cd into /home/jovyan AFTER the mount happens. Something like this may work... I'll explore in the future.

```
CMD tini -- /bin/bash -c "cd /home/jovyan && jupyterhub-singleuser \
  --port=8888 \
  --ip=0.0.0.0 \
  --user="$JPY_USER" \
  --cookie-name=$JPY_COOKIE_NAME \
  --base-url=$JPY_BASE_URL \
  --hub-prefix=$JPY_HUB_PREFIX \
  --hub-api-url=$JPY_HUB_API_URL"
```

**How to test**
- Start with a clean deployment. That means delete all pods, pvc's, pv's, etc...
- Run deployment and make sure you can launch a notebook and do stuff and save file
- Delete user pod
- Launch user pod and make sure files are still persisted.

### CODE REVIEW

File: hub/jupyterhub_config.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 c.KubeSpawner.start_timeout = 60 * 20
 
 # Our simplest user image! Optimized to just... start, and be small!
-c.KubeSpawner.singleuser_image_spec = 'data8/jupyterhub-k8s-user:master'
+c.KubeSpawner.singleuser_image_spec = 'data8/jupyterhub-k8s-user:fix_tini'
+c.KubeSpawner.singleuser_image_pull_policy = 'Always'
```
Comment (derrickmar) [inline hub/jupyterhub_config.py]: Change these after pr is accepted

**Review by SamLau95 — APPROVED** [review]
Nice detective work! Let's get this working for now; we'll prolly have to reach out for help to get it the way we want it.

**Comment by SamLau95** [comment]
@derrickmar Does it sound reasonable to merge these changes now to get things working for the time being? We should change the image tag and pull policy if so.

**Comment by derrickmar** [comment]
Yeah @SamLau95 I'm doing a little more investigating but I'll get this in before tomorrow

**Comment by yuvipanda** [comment]
This has been folded in by other commits

**Pull request closed without merging**